modules/contact/src/actions/ContactAction.C

```diff
@@ -336,19 +363,25 @@ ContactAction::addMortarContact()
     // Add the lagrange multiplier on the secondary subdomain.
     const auto addLagrangeMultiplier =
         [this, &secondary_subdomain_name, &displacements](const std::string & variable_name,
-                                                          const int codimension,
+                                                          const int min_lm_order,
+                                                          const bool frictional,
                                                           const Real scaling_factor) //
     {
       InputParameters params = _factory.getValidParams("MooseVariableBase");
-      params.set<bool>("use_dual") = _use_dual;
+
+      // Allow the user to select "weighted" constraints and standard bases (use_dual = false)
+      // Unless it's for testing purposes, this combination isn't recommended
+      params.set<bool>("use_dual") = (_mortar_approach == MortarApproach::Weighted && _use_dual);
```

Comment:
So if a `ContactAction` user specifies `use_dual` in their input but they're not using `MortarApproach::Weighted`, then their choice is not honored? Is this a change in behavior?

Reply:
It is a change of behavior that isn't present in tests apparently. I made changes to avoid this change of behavior.